backend/fms_core/resources/project_link_sample.py

```diff
@@ -0,0 +1,104 @@
+        # Get sample object
+        try:
+            obj.sample = Sample.objects.get(name=data["Sample Name"],
+                                            container__barcode=data["Sample Container Barcode"],
+                                            coordinates=data["Sample Container Coord"])
+        except Sample.DoesNotExist:
+            obj.sample = None
+            errors["sample"] = ValidationError([f"No sample with name [{data['Sample Name']}] at those container coordinates."], code="invalid")
+        # Get project object
+        try:
+            obj.project = Project.objects.get(name=data["Project Name"])
+        except Project.DoesNotExist:
+            obj.project = None
+            errors["project"] = ValidationError([f"No project with name [{data['Project Name']}]"], code="invalid")
+
+        # Set object for deletion if REMOVE_ACTION
+        if SampleByProject.objects.filter(sample=obj.sample, project=obj.project).exists():
+            # Perform the desired action
+            if data["Action"] == REMOVE_ACTION:
+                    obj.deleted = True
+            elif data["Action"] == ADD_ACTION:
+                errors["link"] = ValidationError(
+                    [f"Sample [{data['Sample Name']}] is already associated to project [{data['Project Name']}]."],
+                    code="invalid")
+
+        else:
+            if data["Action"] == REMOVE_ACTION:
+                errors["link"] = ValidationError([f"Sample [{data['Sample Name']}] is not currently associated to project [{data['Project Name']}]."],
+                                                 code="invalid")
+
+        if errors:
+            raise ValidationError(errors)
+
+    def import_field(self, field, obj, data, is_m2m=False):
+        if field.column_name == "Action":
+            # Normalize Action attribute to be uppercase
+            data["Action"] = get_normalized_str(data, "Action").upper()
+        elif field.column_name == "Project Name":
+            data["Project Name"] = get_normalized_str(data, "Project Name")
+        elif field.column_name == "Sample Name":
+            data["Sample Name"] = get_normalized_str(data, "Sample Name")
+        elif field.column_name == "Sample Container Barcode":
+            data["Sample Container Barcode"] = get_normalized_str(data, "Sample Container Barcode")
+        elif field.column_name == "Sample Container Coord":
+            data["Sample Container Coord"] = get_normalized_str(data, "Sample Container Coord").upper()
+        super().import_field(field, obj, data, is_m2m)
+
+    def after_save_instance(self, instance, using_transactions, dry_run):
+        super().after_save_instance(instance, using_transactions, dry_run)
+        # If the instance is deleted (saved version), delete the instance
+        instance.refresh_from_db()
+        if instance.deleted:
+            super().instance.delete() # Delete without creating a new deleted version
```

Comment:
This will be calling our custom delete() method in tracked_model if your entity model inherits from TrackedModel.
I wonder if there is any special handling of the 'deleted' objects; this is a custom attribute, so Django for now does not what to do with those objects.

i.e: you call: Container.objects..get(id=1).delete() it will call the delete() custom method in our tracked_model.py file. The attribute 'deleted' in our Container row will be set to 'true'
However, if you lookup Container.objects.all(), unless the 'deleted' objects are handled, the Container id 1 will still show up in your list of Container objects. 
Did we think/implemented a mechanism for that?